lib/index.js

```diff
@@ -1,10 +1,18 @@
 'use strict';
 
-var Hoek = require('hoek');
+var Hoek = require('@hapi/hoek');
 var pkg = require('../package.json'); // require package.json for attributes
 
+/**
+ * isFunction checks if a given value is a function.
+ * @param {Object} functionToCheck - the object we want to confirm is a function
+ * @returns {Boolean} true|false
+ */
 function isFunction(functionToCheck) {
-  return functionToCheck && {}.toString.call(functionToCheck) === '[object Function]';
+  const toString = Object.prototype.toString;
+  return functionToCheck
+    && toString.call(functionToCheck) === '[object Function]'
+    || toString.call(functionToCheck) === '[object AsyncFunction]';
```

Comment:
💪

Reply:
Borrowed from [dwyl/hapi-auth-jwt2/lib/index.js#L56](https://github.com/dwyl/hapi-auth-jwt2/blob/490f2fe7761818e5cb118c8cfe44b019cf7d35b4/lib/index.js#L56) not worth creating/using an NPM module for this but worth documenting it with a JSDOC comment. 😉

Reply:
I didn't know about this option, or forgot - [object AsyncFunction]